internal/controllers/inventory_controller.go

```diff
@@ -1541,6 +1600,47 @@ func (t *reconcilerTask) deployServer(ctx context.Context, serverName string) (u
 		},
 	}
 
+	if utils.IsOAuthEnabled(t.object) && utils.RequiresInternalProxy(serverName) {
+		deploymentSpec.Template.Spec.Containers = append(deploymentSpec.Template.Spec.Containers, corev1.Container{
+			Name: utils.InternalRbacContainerName,
+			SecurityContext: &corev1.SecurityContext{
+				AllowPrivilegeEscalation: &privilegeEscalation,
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{"ALL"},
+				},
+			},
+			Image:           os.Getenv(utils.KubeRbacProxyImageName),
+			ImagePullPolicy: corev1.PullPolicy(os.Getenv(utils.ImagePullPolicyEnvName)),
+			Args: []string{
+				fmt.Sprintf("--secure-listen-address=0.0.0.0:%d", utils.InternalProxyPort),
+				fmt.Sprintf("--upstream=http://127.0.0.1:%d/", utils.DefaultContainerPort),
+				"--logtostderr=true",
+				"--tls-cert-file=/secrets/tls/tls.crt",
+				"--tls-private-key-file=/secrets/tls/tls.key",
+				"--tls-min-version=VersionTLS12",
+				"--v=10",
+			},
+			Ports: []corev1.ContainerPort{
+				{
+					Name:          utils.InternalServiceTargetPort,
+					Protocol:      corev1.ProtocolTCP,
+					ContainerPort: utils.InternalProxyPort,
+				},
+			},
+			Resources: corev1.ResourceRequirements{
```

Comment:
Any chance we could add a small comment to explain why we have these exact values?

Reply:
ok ... i'll add a note to say that they are copied from the scaffolding

Reply:
fixed